core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -172,15 +172,15 @@ public Query toQuery(QueryParseContext parseContext) throws QueryParsingExceptio
 
     @Override
```

Comment:
Okay so you suggest to prevent NPEs by null checks in doXContent, then fail later in parser. That's okay I think. I just wanted to point out we can't have these null checks in validate() as long as innerParse might return `null` like today. Will post an update on this.